packages/framework-provider-aws/src/library/events-searcher-adapter.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  BoosterConfig,
+  EventEnvelope,
+  EventFilter,
+  EventInterface,
+  EventSearchResponse,
+  Logger,
+  UUID,
+} from '@boostercloud/framework-types'
+import { DynamoDB } from 'aws-sdk'
```

Comment:
This is the adapter that supports all the searching logic for the new events API. 

**This is the core logic in this PR**